vllm/model_executor/models/qwen2_cls.py

```diff
@@ -0,0 +1,157 @@
+from vllm.sequence import IntermediateTensors, PoolerOutput
+
+from .utils import is_pp_missing_parameter
+
+
+class ReLU(nn.Module):
+
+    def __init__(self):
+        super().__init__()
+        self.activation = nn.ReLU()
+
+    def forward(self, input):
+        input, _ = input
+        return self.activation(input)
+
+
+class Qwen2ForSequenceClassification(nn.Module):
+    packed_modules_mapping = {
+        "qkv_proj": [
+            "q_proj",
+            "k_proj",
+            "v_proj",
+        ],
+        "gate_up_proj": [
+            "gate_proj",
+            "up_proj",
+        ],
+    }
+
+    # LoRA specific attributes
+    supported_lora_modules = [
+        "qkv_proj",
+        "o_proj",
+        "gate_up_proj",
+        "down_proj",
+    ]
+    embedding_modules = {}
+    embedding_padding_modules = []
+
+    def __init__(
+        self,
+        config: Qwen2Config,
+        cache_config: Optional[CacheConfig] = None,
+        quant_config: Optional[QuantizationConfig] = None,
+        lora_config: Optional[LoRAConfig] = None,
+    ) -> None:
+        # TODO (@robertgshaw2): see if this can be moved out
+        if (cache_config.sliding_window is not None
+                and hasattr(config, "max_window_layers")):
+            raise ValueError("Sliding window for some but all layers is not "
+                             "supported. This model uses sliding window "
+                             "but `max_window_layers` = %s is less than "
+                             "`num_hidden_layers` = %s. Please open an issue "
+                             "to discuss this feature." % (
+                                 config.max_window_layers,
+                                 config.num_hidden_layers,
+                             ))
+
+        super().__init__()
+
+        self.config = config
+        self.lora_config = lora_config
+
+        self.quant_config = quant_config
+        self.model = Qwen2Model(config, cache_config, quant_config)
+
+        self.score = RowParallelLinear(config.hidden_size,
+                                 config.num_labels,
+                                 quant_config=quant_config)
+        self._pooler = Pooler(pooling_type=PoolingType.LAST, normalize=False, softmax=True)
+
+    def forward(
+        self,
+        input_ids: torch.Tensor,
+        positions: torch.Tensor,
+        kv_caches: List[torch.Tensor],
+        attn_metadata: AttentionMetadata,
+        intermediate_tensors: Optional[IntermediateTensors] = None,
+    ) -> torch.Tensor:
+        hidden_states = self.model(input_ids, positions, kv_caches,
+                                   attn_metadata, intermediate_tensors)
+        logits, _ = self.score(hidden_states)
+        return logits
+
+    def pooler(
+        self,
+        hidden_states: torch.Tensor,
+        pooling_metadata: PoolingMetadata,
+    ) -> Optional[PoolerOutput]:
+        return self._pooler(hidden_states, pooling_metadata)
```

Comment:
I think it makes more sense to place the softmax here instead of inside pooler.

Reply:
Thank you for your opinion. 
I agree with your way.

I'll commit this with followed by yours.

Reply:
@DarkLight1337  this cause the runtime test failure.
Do you have any idea about this?

Reply:
I think it's because `PoolerOutput` itself is not a tensor. Hmm... maybe we have to revert back to applying softmax inside the pooler.

Reply:
@DarkLight1337 I see. I'll do this now

Reply:
@DarkLight1337  Just pushed. Lets waiting the test results.